src/components/FlatList/index.android.tsx

```diff
@@ -0,0 +1,64 @@
+import {useFocusEffect} from '@react-navigation/native';
+import React, {ForwardedRef, forwardRef, useCallback, useState} from 'react';
+import {FlatList, FlatListProps} from 'react-native';
+
+type CustomFlatListProps = {
+    /** Same as for FlatList */
+    onScroll?: (...args: unknown[]) => unknown;
+
+    /** Same as for FlatList */
+    onLayout?: () => (...args: unknown[]) => unknown;
+
+    /** Passed via forwardRef so we can access the FlatList ref */
+    innerRef: ForwardedRef<FlatList>;
+} & FlatListProps<unknown>;
```

Comment:
I think you can use generics like @VickyStash used [here](https://github.com/Expensify/App/pull/31065/files), wdyt?

Also, let's create a `types.ts` file to store this props type.